File: templates/Pages/MediaPlayerVB/ViewModels/MediaPlayerViewViewModel.vb
Imports Windows.Media.Playback
Imports Windows.Media.Core

Namespace ViewModels
    Public Class MediaPlayerViewViewModel
        Inherits System.ComponentModel.INotifyPropertyChanged

        ' TODO WTS: Specify your video default and image here
        Private Const DefaultSource As String = "https://sec.ch9.ms/ch9/db15/43c9fbed-535e-4013-8a4a-a74cc00adb15/C9L12WinTemplateStudio_high.mp4"

        ' The poster image is displayed until the video is started
        Private Const DefaultPoster As String = "https://sec.ch9.ms/ch9/db15/43c9fbed-535e-4013-8a4a-a74cc00adb15/C9L12WinTemplateStudio_960.jpg"

        Private _source As IMediaPlaybackSource

        Public Property Source As IMediaPlaybackSource
            Get
                Return _source
            End Get
            Set
                [Param_Setter](_source, value)
            End Set
        End Property

        Private _posterSource As String

        Public Property PosterSource As String
            Get
                Return _posterSource
            End Get
            Set
                [Param_Setter](_posterSource, newValue := value)
            End Set
        End Property

        Public Sub New()
            Source = MediaSource.CreateFromUri(New Uri(DefaultSource))
            PosterSource = DefaultPoster
        End Sub
    End Class
End Namespace

File: code/TemplateStudioForUWP/Templates/Pg/MediaPlayer/ViewModels/MediaPlayerViewViewModel.cs
using System;
using Windows.Media.Playback;
using Windows.Media.Core;

namespace Param_RootNamespace.ViewModels
{
    public class MediaPlayerViewViewModel : System.ComponentModel.INotifyPropertyChanged
    {
        // TODO: Set your default video and image URIs
        private const string DefaultSource = "https://sec.ch9.ms/ch9/db15/43c9fbed-535e-4013-8a4a-a74cc00adb15/C9L12WinTemplateStudio_high.mp4";

        // The poster image is displayed until the video is started
        private const string DefaultPoster = "https://sec.ch9.ms/ch9/db15/43c9fbed-535e-4013-8a4a-a74cc00adb15/C9L12WinTemplateStudio_960.jpg";

        private IMediaPlaybackSource _source;

        public IMediaPlaybackSource Source
        {
            get { return _source; }
            set { Param_Setter(ref _source, value); }
        }

        private string _posterSource;

        public string PosterSource
        {
            get { return _posterSource; }
            set { Param_Setter(ref _posterSource, value); }
        }

        public MediaPlayerViewViewModel()
        {
            Source = MediaSource.CreateFromUri(new Uri(DefaultSource));
            PosterSource = DefaultPoster;
        }

        public void DisposeSource()
        {
            var mediaSource = Source as MediaSource;
            mediaSource?.Dispose();
            Source = null;
        }
    }
}
